examples/gimli/src/gimli.rs

```diff
@@ -121,7 +121,8 @@ fn process_msg(message: &ByteSeq, mut s: State) -> (State, ByteSeq) {
     let mut ciphertext = ByteSeq::new(message.len());
 
     let rate = Block::length();
-    let num_chunks = message.num_chunks(rate);
+    let num_chunks = message.num_exact_chunks(rate) + 1;
```

Comment:
Hm, I don't like the overrunning. I think we can do this with exact chunks without too much code duplication. The following should work. What do you think?

```rust
    let rate = Block::length();
    let num_chunks = message.num_exact_chunks(rate);

    for i in 0..num_chunks {
        let key_block = squeeze_block(s);
        let (_block_len, msg_block) = message.get_chunk(rate, i);
        let msg_block = Block::from_seq(&msg_block);
        ciphertext = ciphertext.set_exact_chunk(rate, i, &(msg_block ^ key_block));
        s = absorb_block(msg_block, s);
    }

    let key_block = squeeze_block(s);
    let last_block = message.get_remainder_chunk(rate);
    let block_len = last_block.len();
    let msg_block_padded = Block::new();
    let mut msg_block_padded = msg_block_padded.update_start(&last_block);

    ciphertext = ciphertext.set_chunk(
        rate,
        num_chunks,
        // the slice_range cuts off the last block if it is padded
        &(msg_block_padded ^ key_block).slice_range(0..block_len),
    );
    msg_block_padded[block_len] = msg_block_padded[block_len] ^ U8(1u8);
    s[11] = s[11] ^ U32(0x01000000u32); // s_2,3
    s = absorb_block(msg_block_padded, s);
```

Reply:
Okay done!